test/java/seedu/addressbook/data/AddressBookTest.java

```diff
@@ -0,0 +1,275 @@
+        UniqueTagList tags;
+        UniquePersonList persons;
+        AddressBook addressBook;
+        
+        try {
+            tags = new UniqueTagList(tagMathematician, tagScientist);
+            persons = new UniquePersonList(aliceBetsy, bobChaplin);
+            addressBook = new AddressBook(persons, tags);    
+        } catch (DuplicateTagException | DuplicatePersonException e) {
+            // The choice of Tag and Person values are such that this method never throws.
+            // We should thus never reach this point.
+            throw new RuntimeException();
+        }
+
+        return addressBook;
+    }
+    
+    @Before
+    public void setUp() throws Exception {
+        tagPrizeWinner   = new Tag("prizewinner");
+        tagScientist     = new Tag("scientist");
+        tagMathematician = new Tag("mathematician");
+        tagEconomist     = new Tag("economist");
+
+        aliceBetsy     = new Person(new Name("Alice Betsy"),
+                                    new Phone("91235468", false),
+                                    new Email("alice@nushackers.org", false),
+                                    new Address("8 Computing Drive, Singapore", false),
+                                    new UniqueTagList(tagMathematician));
+
+        bobChaplin     = new Person(new Name("Bob Chaplin"),
+                                    new Phone("94321500", false),
+                                    new Email("bob@nusgreyhats.org", false),
+                                    new Address("9 Computing Drive", false),
+                                    new UniqueTagList(tagMathematician));
+
+        charlieDouglas = new Person(new Name("Charlie Douglas"),
+                                    new Phone("98751365", false),
+                                    new Email("charlie@nusgdg.org", false),
+                                    new Address("10 Science Drive", false),
+                                    new UniqueTagList(tagScientist));
+
+        davidElliot    = new Person(new Name("David Elliot"),
+                                    new Phone("84512575", false),
+                                    new Email("douglas@nuscomputing.com", false),
+                                    new Address("11 Arts Link", false),
+                                    new UniqueTagList(tagEconomist, tagPrizeWinner));
+    }
+    
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void syncTagsWithMasterList_emptyInitialTagList_addsAllIntoTagList() throws Exception {
+        UniqueTagList emptyTagList = new UniqueTagList();
+        UniquePersonList persons = new UniquePersonList(bobChaplin, charlieDouglas);
+
+        AddressBook addressBook = new AddressBook(persons, emptyTagList);
+
+        assertTrue(addressBook.containsTag(tagMathematician));
+        assertTrue(addressBook.containsTag(tagScientist));
+        assertFalse(addressBook.containsTag(tagEconomist));
```

Comment:
Can this be done in one shot (and a more definitive way)? e.g. 
`assertEquals(expectedTagList, addressBook.getAllTags())`